LOB test fails on server which provides multiple implementations of XML Binding

4 tests in `org.eclipse.persistence.testing.tests.jpa.lob.LobJUnitTestCase` are failing:

```
[ERROR] Errors: 
[ERROR]   LobJUnitTestCase.testCreate » EJB java.nio.channels.ClosedChannelException
[ERROR]   LobJUnitTestCase>JUnitTestCase.runBareServer:825->testDelete:91 » IllegalArgument Object: null is not a known Entity type.
[ERROR]   LobJUnitTestCase>JUnitTestCase.runBareServer:825->testRead:110 NullPointer
[ERROR]   LobJUnitTestCase.testSetup » EJB java.nio.channels.ClosedChannelException
[ERROR]   LobJUnitTestCase>JUnitTestCase.runBareServer:825->testUpdate:141 NullPointer
```

to reproduce, run following on the server, ie:
`mvn test -pl :org.eclipse.persistence.jpa.testapps.lob -Pmysql,wildfly -Dserver.test.skip=false`

the cause is:
```
...
Caused by: java.lang.RuntimeException: jakarta.xml.bind.PropertyException: name: eclipselink.media-type value: application/json
	at org.eclipse.persistence@27.0.0.Alpha2//org.eclipse.persistence.sessions.serializers.JSONSerializer.serialize(JSONSerializer.java:53)
	at org.eclipse.persistence@27.0.0.Alpha2//org.eclipse.persistence.mappings.converters.SerializedObjectConverter.convertObjectValueToDataValue(SerializedObjectConverter.java:162)
	... 127 more
Caused by: jakarta.xml.bind.PropertyException: name: eclipselink.media-type value: application/json
	at jakarta.xml.bind.api@4.0.0//jakarta.xml.bind.helpers.AbstractMarshallerImpl.setProperty(AbstractMarshallerImpl.java:355)
	at org.glassfish.jaxb//org.glassfish.jaxb.runtime.v2.runtime.MarshallerImpl.setProperty(MarshallerImpl.java:492)
	at org.eclipse.persistence@27.0.0.Alpha2//org.eclipse.persistence.sessions.serializers.JSONSerializer.serialize(JSONSerializer.java:48)
	... 128 more
```

--- a/foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/sessions/serializers/XMLSerializer.java
+++ b/foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/sessions/serializers/XMLSerializer.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 1998, 2021 Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 1998, 2022 Oracle and/or its affiliates. All rights reserved.
  *
  * This program and the accompanying materials are made available under the
  * terms of the Eclipse Public License v. 2.0 which is available at
@@ -16,19 +16,24 @@ package org.eclipse.persistence.sessions.serializers;
 
 import java.io.StringReader;
 import java.io.StringWriter;
+import java.util.Map;
 
 import jakarta.xml.bind.JAXBContext;
 import jakarta.xml.bind.JAXBException;
 import jakarta.xml.bind.Marshaller;
 import jakarta.xml.bind.Unmarshaller;
 
+import org.eclipse.persistence.internal.security.PrivilegedAccessHelper;
 import org.eclipse.persistence.sessions.Session;
 
 /**
- * Uses JAXB to convert an object to XML.
+ * Uses JAXB (MOXy runtime) to convert an object to XML.
  * @author James Sutherland
  */
 public class XMLSerializer extends AbstractSerializer {
+
+    private static final Map<String, Object> XML_FACTORY_OPTS = Map.of(JAXBContext.JAXB_CONTEXT_FACTORY, "org.eclipse.persistence.jaxb.XMLBindingContextFactory");
+
     JAXBContext context;
 
     public XMLSerializer() {
@@ -36,7 +41,9 @@ public class XMLSerializer extends AbstractSerializer {
 
     public XMLSerializer(String packageName) {
         try {
-            this.context = JAXBContext.newInstance(packageName);
+            this.context = JAXBContext.newInstance(packageName,
+                    PrivilegedAccessHelper.callDoPrivileged(() -> Thread.currentThread().getContextClassLoader()),
+                    XML_FACTORY_OPTS);
         } catch (JAXBException exception) {
             throw new RuntimeException(exception);
         }
@@ -53,7 +60,7 @@ public class XMLSerializer extends AbstractSerializer {
                 serializePackage = serializeClass.getPackage().getName();
             }
             try {
-                this.context = JAXBContext.newInstance(serializePackage, serializeClass.getClassLoader());
+                this.context = JAXBContext.newInstance(serializePackage, serializeClass.getClassLoader(), XML_FACTORY_OPTS);
             } catch (JAXBException exception) {
                 throw new RuntimeException(exception);
             }

--- a/jpa/eclipselink.jpa.testapps/jpa.test.lob/pom.xml
+++ b/jpa/eclipselink.jpa.testapps/jpa.test.lob/pom.xml
@@ -46,8 +46,6 @@
 
     <properties>
         <argLine/>
-        <!-- server tests require MOXy as the default XML Binding provider -->
-        <server.test.skip>true</server.test.skip>
     </properties>
 
     <build>

--- a/jpa/eclipselink.jpa.testapps/jpa.test.lob/src/test/java/org/eclipse/persistence/testing/tests/jpa/lob/LobTest.java
+++ b/jpa/eclipselink.jpa.testapps/jpa.test.lob/src/test/java/org/eclipse/persistence/testing/tests/jpa/lob/LobTest.java
@@ -30,29 +30,29 @@ import java.lang.reflect.Field;
 /**
  * JUnit test case(s) for the TopLink EntityMappingsXMLProcessor.
  */
-public class LobJUnitTestCase extends JUnitTestCase {
+public class LobTest extends JUnitTestCase {
 
     private static Integer imageId;
     private static Image originalImage;
 
-    public LobJUnitTestCase() {
+    public LobTest() {
         super();
     }
 
-    public LobJUnitTestCase(String name) {
+    public LobTest(String name) {
         super(name);
     }
 
     public static Test suite() {
         TestSuite suite = new TestSuite("Lob Model");
-        suite.addTest(new LobJUnitTestCase("testSetup"));
-        suite.addTest(new LobJUnitTestCase("testMerge"));
-        suite.addTest(new LobJUnitTestCase("testCreate"));
-        suite.addTest(new LobJUnitTestCase("testRead"));
-        suite.addTest(new LobJUnitTestCase("testUpdate"));
-        suite.addTest(new LobJUnitTestCase("testDelete"));
-        suite.addTest(new LobJUnitTestCase("testMerge"));
-        suite.addTest(new LobJUnitTestCase("testRetrieveLazyBasicAfterTxnReadCommit"));
+        suite.addTest(new LobTest("testSetup"));
+        suite.addTest(new LobTest("testMerge"));
+        suite.addTest(new LobTest("testCreate"));
+        suite.addTest(new LobTest("testRead"));
+        suite.addTest(new LobTest("testUpdate"));
+        suite.addTest(new LobTest("testDelete"));
+        suite.addTest(new LobTest("testMerge"));
+        suite.addTest(new LobTest("testRetrieveLazyBasicAfterTxnReadCommit"));
 
         return suite;
     }
@@ -250,6 +250,6 @@ public class LobJUnitTestCase extends JUnitTestCase {
     }
 
     public static void main(String[] args) {
-        junit.textui.TestRunner.run(LobJUnitTestCase.suite());
+        junit.textui.TestRunner.run(LobTest.suite());
     }
 }

--- a/jpa/org.eclipse.persistence.jpa/pom.xml
+++ b/jpa/org.eclipse.persistence.jpa/pom.xml
@@ -42,6 +42,12 @@
         </dependency>
         <dependency>
             <groupId>org.eclipse.persistence</groupId>
+            <artifactId>org.eclipse.persistence.moxy</artifactId>
+            <scope>runtime</scope>
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <groupId>org.eclipse.persistence</groupId>
             <artifactId>org.eclipse.persistence.json</artifactId>
             <scope>runtime</scope>
             <optional>true</optional>
@@ -73,6 +79,12 @@
         <dependency>
             <groupId>jakarta.validation</groupId>
             <artifactId>jakarta.validation-api</artifactId>
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <groupId>jakarta.activation</groupId>
+            <artifactId>jakarta.activation-api</artifactId>
+            <scope>provided</scope>
             <optional>true</optional>
         </dependency>
         <dependency>

